File: algorithms/warmup/compare_the_triplets.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	aliceInput, _ := reader.ReadString('\n')
	aliceInput =  strings.TrimSpace(aliceInput)
	bobInput, _ := reader.ReadString('\n')
	bobInput =  strings.TrimSpace(bobInput)

	aliceRates := strings.Split(aliceInput, " ")
	bobRates := strings.Split(bobInput, " ")

	aliceRating := 0
	bobRating := 0

	for index, value := range aliceRates {
		aliceValue, _ := strconv.Atoi(value)
		bobValue, _ := strconv.Atoi(bobRates[index])

		if aliceValue > bobValue {
			aliceRating++
		}

		if aliceValue < bobValue {
			bobRating++
		}
	}

	result := strings.Join([]string{
		strconv.Itoa(aliceRating),
		strconv.Itoa(bobRating),
	}, " ")

	fmt.Println(result)
}

File: algorithms/warmup/compare_the_triplets.php
<?php
declare(strict_types=1);

/**
 * @param int $a0
 * @param int $a1
 * @param int $a2
 * @param int $b0
 * @param int $b1
 * @param int $b2
 * @return int[]
 */
function solve(int $a0, int $a1, int $a2, int $b0, int $b1, int $b2): array
{
    $aliceRating = 0;
    $bobRating = 0;
    $aliceRates = [$a0, $a1, $a2];
    $bobRates = [$b0, $b1, $b2];

    foreach ($aliceRates as $key => $aliceRate) {
        $bobRate = $bobRates[$key];

        if ($aliceRate > $bobRate) {
            $aliceRating++;
        }

        if ($aliceRate < $bobRate) {
            $bobRating++;
        }
    }

    return [$aliceRating, $bobRating];
}

$handle = fopen ("php://stdin", "r");
fscanf($handle, "%d %d %d", $a0, $a1, $a2);
fscanf($handle, "%d %d %d", $b0, $b1, $b2);

$result = solve($a0, $a1, $a2, $b0, $b1, $b2);
echo implode(" ", $result) . "\n";
